chanzuckerberg/idseq-web
# New host organism input

# Description

![image](https://user-images.githubusercontent.com/28797/73307933-06223a00-41d4-11ea-90b2-0a2465c2551e.png)

This changes the host genome input to be like the new sample type input. It is searchable and accepts plain text, which become new host genomes on submit, defaulting to ERCC only subtraction. (See #2968 for messaging to users on review step. And see #2987 for how to distinguish downstream.)

![image](https://user-images.githubusercontent.com/28797/73308024-32d65180-41d4-11ea-9639-d088632048aa.png)

I also updated the top description according to the new design by @happyimadesignr .

![image](https://user-images.githubusercontent.com/28797/73389585-57d8cc00-4289-11ea-81c1-ccf3cecf6e4c.png)

I also added a "ERCC subtraction only" subtext to the dropdown according to design review.

![image](https://user-images.githubusercontent.com/28797/73394576-bc4c5900-4292-11ea-8653-0738f835dd39.png)


# Notes

* Feature is admin only gated
* Not sure if our QA team has admin accounts to test this in staging
* New host genomes simply point to the ERCC index files by new defaults in previous PR
* Invalid names such as "1234" will return an error
* The dropdown order is initially by popularity, then by text match rank, same as sample type
* I added a behavior to the dropdown: exact match eliminates the plain text option
* I will perform a global replace of all UI strings to "host organism" in a later PR when we are ready to launch. 

# Tests

* new and old unit tests
* try upload as admin, see new input, complete steps
* try upload as non admin, see old input, complete steps
* try invalid name, see error message

![image](https://user-images.githubusercontent.com/28797/73310574-6667aa80-41d9-11ea-9699-b44af856251b.png)
